### PR TITLE
feat(CC): add data source cloud connections

### PR DESCRIPTION
**Pull request by profoundwu**
**What this PR does / why we need it**:
add data source cloud connections
**Which issue this PR fixes**:
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
fixes #xxx

**Special notes for your reviewer**:

**Release note**:

```release-note
1. add data source cloud connections
```

## PR Checklist

* [x] Tests added/passed.
* [x] Documentation updated.
* [x] Schema updated.

## Acceptance Steps Performed

```
 make testacc TEST="./huaweicloud/services/acceptance/cc" TESTARGS="-run TestAccDataSourceCloudConnections_basic"
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/cc -v -run TestAccDataSourceCloudConnections_basic -timeout 360m -parallel 4
=== RUN   TestAccDataSourceCloudConnections_basic
=== PAUSE TestAccDataSourceCloudConnections_basic
=== CONT  TestAccDataSourceCloudConnections_basic
--- PASS: TestAccDataSourceCloudConnections_basic (58.31s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/cc        58.363s
```

### CODE REVIEW

File: docs/data-sources/cc_connections.md [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+```
+
+## Argument Reference
+
+The following arguments are supported:
+
+* `region` - (Optional, String) Specifies the region in which to query the resource.
+  If omitted, the provider-level region will be used.
+
+* `cloud_connection_id` - (Optional, String) Specifies the cloud connection ID.
```
Comment (Jason-Zhang9309) [inline docs/data-sources/cc_connections.md]:
```suggestion
* `connection_id` - (Optional, String) Specifies the cloud connection ID.
```

File: docs/data-sources/cc_connections.md [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+
+* `tags` - (Optional, Map) Specifies the cloud connection tags.
+
+## Attribute Reference
+
+In addition to all arguments above, the following attributes are exported:
+
+* `id` - The data source ID.
+
+* `cloud_connections` - Cloud connection list.
```
Comment (Jason-Zhang9309) [inline docs/data-sources/cc_connections.md]:
```suggestion
* `connections` - Cloud connection list.
```

File: huaweicloud/services/cc/data_source_huaweicloud_cc_connections.go [hunk 3]
```diff
@@ -0,0 +1,324 @@
...
+	"github.com/chnsz/golangsdk"
+	"github.com/chnsz/golangsdk/pagination"
+
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/config"
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/utils"
+)
+
+// @API CC GET /v3/{domain_id}/ccaas/cloud-connections
+// @API CC POST /v3/{domain_id}/ccaas/cloud-connections/filter
+func DataSourceCcCloudConnections() *schema.Resource {
```
Comment (Jason-Zhang9309) [inline huaweicloud/services/cc/data_source_huaweicloud_cc_connections.go]: other func names alse need to updated
```suggestion
func DataSourceCloudConnections() *schema.Resource {
```

File: docs/data-sources/cc_connections.md (hunk 1 above)
Comment (profoundwu) [reply docs/data-sources/cc_connections.md]: fixed

File: huaweicloud/services/cc/data_source_huaweicloud_cc_connections.go (hunk 3 above)
Comment (profoundwu) [reply huaweicloud/services/cc/data_source_huaweicloud_cc_connections.go]: fixed

File: docs/data-sources/cc_connections.md (hunk 2 above)
Comment (profoundwu) [reply docs/data-sources/cc_connections.md]: fixed

**Pull request merged**